frontend/src/components/admin/question-creation/question-elements/modals/fraction/AddFractionModal.tsx

```diff
@@ -1,20 +1,24 @@
-import React, { useContext } from "react";
+import React, { useContext, useState } from "react";
+import { useDisclosure } from "@chakra-ui/react";
 import { v4 as uuidv4 } from "uuid";
 
 import QuestionEditorContext from "../../../../../../contexts/QuestionEditorContext";
 import type { FractionMetadata } from "../../../../../../types/QuestionMetadataTypes";
+import type { FractionType } from "../../../../../../types/QuestionTypes";
 import { QuestionElementType } from "../../../../../../types/QuestionTypes";
 
+import ChooseFractionTypeModal from "./ChooseFractionTypeModal";
 import FractionModal from "./FractionModal";
 
 const AddFractionModal = (): React.ReactElement => {
-  const { showAddFractionModal, setShowAddFractionModal } = useContext(
-    QuestionEditorContext,
-  );
-  const closeModal = () => {
-    setShowAddFractionModal(false);
-  };
+  const [fractionType, setFractionType] = useState<FractionType>("regular");
+  const {
+    onOpen: onSecondModalOpen,
```

Comment:
nit: instead of referring to `first` and `second` modal, could we do `ChooseFraction` and `FractionInput` or something like that instead?